### PR TITLE
The database has been improved.

### PR DESCRIPTION
**Pull request by vitderkach**
Location and City tables have been added; vacation types have been rewritten; RecruitmentDate and IsFired fields to UserInformation table have been added; VacationPolicy table has been got a support keep data for different years; also some cosmetic improvements have been  applied.

### CODE REVIEW

**Comment by vitderkach** [comment]
I can rename the migration, But in my opinion, spliting the migration into several parts is a good idea. The migration is just a way to see how database versions have changed, and, if it's nedded, to revert them. So, if we even so split the migration, it does mean that we had more versions of the database than it is real.
And more, the size of migration code does't depend from me. It's done using npm's third-part tool, which will generate code automatically. I must delete all done changes and re-add the tables consistently to do what you want.

**Comment by vitderkach** [comment]
The administrative leave is an unpaid temporary leave.

**Comment by vitderkach** [comment]
Снимок экрана (180)

**Comment by mishavetl** [comment]
> The administrative leave is an unpaid temporary leave.

In Excel file, it counts as a separately, so it should be stored separately.

**Comment by vitderkach** [comment]
As I see in the excel file, nowadays we have five types of leave. Where you saw the in the file is an "unpaid leave", I don't know. Nowadays we have in enum all of that types of vacations plus "gift leave", as it needs according the task, and  "for unknown reasons" until the admin doesn't a reason why the employee missed the a workday.
Снимок экрана (182)
Снимок экрана (183)

File: TOT.Business/Services/ManagerService.cs [hunk 1]
```diff
@@ -219,141 +220,147 @@ public ManagerResponseDto GetResponseByVacationId(int vacationRequestId)
             }
         }
 
-        // TODO: Review the method
-        #region CaluclateVacationDays
+        // TODO: Rewrite the mehod because the database logic has been changed. As an example the commented code below
```
Comment (mishavetl) [inline TOT.Business/Services/ManagerService.cs]: Fix a typo 'Rewrite the __method__ because...'.

File: TOT.Business/Services/VacationService.cs [hunk 2]
```diff
@@ -73,17 +74,24 @@ public void ApplyForVacation(VacationRequestDto vacationRequestDto)
...
-            var vacationDays = _uow.VacationPolicyRepository
-                .GetAll()
-                .Where(v => v.UserInformation.ApplicationUser.Id == userId)
-                .FirstOrDefault();
-            var vacationDaysDto = _mapper.Map<VacationPolicy, VacationDaysDto>(vacationDays);
-            return vacationDaysDto;
+            throw new NotImplementedException();
         }
 
+        // TODO: Rewrite the mehod because the database logic has been changed. As an example the commented code below
```
Comment (mishavetl) [inline TOT.Business/Services/VacationService.cs]: The same typo here)

**Comment by mishavetl** [comment]
I have a problem starting application locally.

```
fail: Microsoft.EntityFrameworkCore.Database.Command[20102]
      Failed executing DbCommand (19ms) [Parameters=[], CommandType='Text', CommandTimeout='30']

                      ALTER TABLE dbo.VacationRequests
                      ADD SystemStart DATETIME2(0)
                      GENERATED ALWAYS AS ROW START HIDDEN NOT NULL
                      CONSTRAINT DFT_VacationRequests_SystemStart DEFAULT('19000101'),
                      SystemEnd DATETIME2(0)
                      GENERATED ALWAYS AS ROW END HIDDEN NOT NULL
                      CONSTRAINT DFT_VacationRequests_SystemEnd  DEFAULT('99991231 23:59:59'),
                      PERIOD FOR SYSTEM_TIME(SystemStart, SystemEnd);
                      ALTER TABLE dbo.VacationRequests
                      SET(SYSTEM_VERSIONING = ON(HISTORY_TABLE = History.VacationRequestsHistory));
System.Data.SqlClient.SqlException (0x80131904): Setting SYSTEM_VERSIONING to ON failed because table 'TimeOffTrackerDb.dbo.VacationRequests' has a FOREIGN KEY with cascading DELETE or UPDATE.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
```

**Comment by vitderkach** [comment]
You need to have 2017 Sql Server exmaple or newer. It doesn't depend from me.

**Pull request merged**